sql/set.go

```diff
@@ -39,7 +39,12 @@ func (p *planner) Set(n *parser.Set) (planNode, error) {
 		if err != nil {
 			return nil, err
 		}
-		p.session.Database = val.String()
+		if s, ok := val.(parser.DString); ok {
+			p.session.Database = string(s)
+		} else if !ok {
+			return nil, fmt.Errorf("database: requires a single string value: %s is a %s",
+				n.Values[0], val.Type())
```

Comment:
I wanted the type when I was debugging this. I suppose an argument could be made either way.